### PR TITLE
docker_image_availability: enable skopeo to use proxies

### PR DESCRIPTION
**Pull request by sosiouxme**
Fixes https://github.com/openshift/openshift-ansible/issues/6300 and https://bugzilla.redhat.com/show_bug.cgi?id=1499358

Run skopeo via shell and add in environment variables for proxies if specified in variables.

### CODE REVIEW

**Comment by sosiouxme** [comment]
unrelated flakes
/retest

**Comment by sosiouxme** [comment]
Maven problem on https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6716/test_pull_request_openshift_ansible_logging/4648/ -- let's see if it's resolved itself before I report a flake

/retest

**Comment by sosiouxme** [comment]
Note there was a [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1491570) for this that should be addressed once this is in good shape.

**Comment by sosiouxme** [comment]
bot, retest this please

**Comment by DanyC97** [comment]
i wished i knew about this as i just spent 1 day on 3.7 trying to understand why .. while refusing to disable the check.

the sooner it gets backported the better

**Comment by sosiouxme** [comment]
@michaelgugino can you look at this if you have time or tag someone else to do so?

File: roles/openshift_health_checker/openshift_checks/docker_image_availability.py [hunk 1]
```diff
@@ -76,11 +76,19 @@ def __init__(self, *args, **kwargs):
...
             oreg_auth_user = self.template_var(oreg_auth_user)
             oreg_auth_password = self.template_var(oreg_auth_password)
-            self.skopeo_command_creds = "--creds={}:{}".format(quote(oreg_auth_user), quote(oreg_auth_password))
+            self.skopeo_command_creds = quote("--creds={}:{}".format(oreg_auth_user, oreg_auth_password))
 
         # record whether we could reach a registry or not (and remember results)
         self.reachable_registries = {}
 
+        # take note of any proxy settings needed
+        proxies = []
```
Comment (michaelgugino) [inline roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: When building variables like this, it's nice to add some comments like
#check openshift_http_proxy openshift_https_proxy openshift_no_proxy
so users/devs can grep for them.

File: roles/openshift_health_checker/openshift_checks/docker_image_availability.py [hunk 2]
```diff
@@ -76,11 +76,19 @@ def __init__(self, *args, **kwargs):
...
+        for var in ['http_proxy', 'https_proxy', 'no_proxy']:
+            value = self.get_var("openshift_" + var, default=None)
```
Comment (michaelgugino) [inline roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: These need to be templated.

File: roles/openshift_health_checker/openshift_checks/docker_image_availability.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ class DockerImageAvailability(DockerHostMixin, OpenShiftCheck):
     # to look for images available remotely without waiting to pull them.
     dependencies = ["python-docker-py", "skopeo"]
     # command for checking if remote registries have an image, without docker pull
-    skopeo_command = "timeout 10 skopeo inspect --tls-verify={tls} {creds} docker://{registry}/{image}"
+    skopeo_command = "{envvars} timeout 10 skopeo inspect --tls-verify={tls} {creds} docker://{registry}/{image}"
```
Comment (michaelgugino) [inline roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: Please rename this to {proxyvars} as those are the only things we're actually setting.

**Comment by sosiouxme** [comment]
Thanks, rebased and addressed review comments.

**Comment by sosiouxme** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6716/test_pull_request_openshift_ansible_extended_conformance_install/4222/ is flake https://github.com/openshift/origin/issues/14296

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6716/test_pull_request_openshift_ansible_extended_conformance_install_crio/1332/ failed in 7 different ways that all seem unrelated to this PR

/retest

**Pull request merged**